asllib/libdir/stdlib0.asl

```diff
@@ -2,30 +2,22 @@
 // Backwards compatibility for ASLv0
 
 bits(N) ReplicateBit(boolean isZero, integer N)
-  return if isZero then Zeros(N) else Ones(N);
+  return ReplicateBit{N}(isZero);
 
 bits(M*N) Replicate(bits(M) x, integer N)
-  if M == 1 then
-    return ReplicateBit(IsZero(x), M*N);
-  else
-    r = Zeros(M*N);
-    for i=0 to N-1
-        r<M*(i+1) - 1 : M*i> = x;
-    return r;
+  return Replicate{M*N}(x);
```

Comment:
Hi @hrutvik I do not understand here, wouldn't the signature de V1 Replicate be `func Replicate{N,M}(x: bits(M)) => bits(N)`...

Reply:
The signature of V1 `Replicate` has changed as you describe:
   ```
   bits(M*N) Replicate(bits(M) x, integer N)    // V0
   func Replicate{N,M}(x: bits(M)) => bits(N)   // V1
   ```
And then I've elided the "input" parameter of the standard library function. Does this seem right to you, or have I missed something somewhere?

Reply:
Well,  my question was not very clear.  Although I have not a very strong opinion on this, I think that giving all parameters in the call would be preferable: `return Replicate{M*N,M}(x);`

Reply:
That makes sense - I've updated to the fully-explicit form as you suggest